File: solutions/java/word-count/1/src/main/java/WordCount.java
import java.util.HashMap;

class WordCount {
    HashMap<String, Integer> wordCount = new HashMap();
    
    HashMap<String, Integer> phrase (String input){
        String[] words = input.split("[ .,\n!&@$%^&=:]+");
        
        for (var word: words){
            if (word == "" || word == "'" || word == "''") continue;
            if (word.charAt(0) == '\'') word = word.substring(1, word.length());
            if (word.charAt(word.length() - 1) == '\'') word = word.substring(0, word.length() - 1);

            word = word.toLowerCase();

            var count = this.wordCount.get(word);
            if (count == null){
                this.wordCount.put(word, 1);
            } else {
                this.wordCount.put(word, count + 1);
            }
        }

        return this.wordCount;
    }
}

File: solutions/javascript/word-count/1/word-count.js
const delimiter = /[^a-zA-Z1-9']+/;

export const countWords = (input) => {
  return input
    .split(delimiter)
    .filter((e) => e != "")
    .map((e) => e.toLowerCase())
    .map((e) =>
      e.charAt(0) == "'" && e.charAt(e.length - 1) == "'"
        ? e.substring(1, e.length - 1)
        : e
    )
    .reduce((acc, cur) => {
      if (acc[cur] == undefined) acc[cur] = 1;
      else acc[cur]++;
      return acc;
    }, {});
};
